lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -276,7 +283,20 @@ protected void updateMinCompetitiveScore(Scorable scorer) throws IOException {
           && pqTop.score != Float.NEGATIVE_INFINITY) { // -Infinity is the score of sentinels
       // since we tie-break on doc id and collect in doc id order, we can require
       // the next float
-      scorer.setMinCompetitiveScore(Math.nextUp(pqTop.score));
+      float bottomScore = Math.nextUp(pqTop.score);
+
+      if (bottomValueChecker != null && bottomValueChecker.getBottomValue() > 0) {
```

Comment:
Why do you need to check that the value is greater than 0 ? It seems that it prevents the updating of the global minimum score since the value is always 0.